Controller/adminhtml/Faq/Delete.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Inchoo\ProductFAQ\Controller\Adminhtml\Faq;
+
+use Inchoo\ProductFAQ\Api\FaqRepositoryInterface;
+use Magento\Backend\App\Action;
+
+class Delete extends Action
+{
+    /**
+     * Delete constructor.
+     * @param Action\Context $context
+     * @param FaqRepositoryInterface $faqRepository
+     */
+    public function __construct(Action\Context $context, FaqRepositoryInterface $faqRepository)
+    {
+        parent::__construct($context);
+        $this->faqRepository = $faqRepository;
+    }
+
+    /**
+     * @return bool
+     */
+    public function _isAllowed()
+    {
+        return $this->_authorization->isAllowed('Inchoo_ProductFAQ::productfaq');
+    }
+
+    /**
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function execute()
+    {
+        $id = $this->getRequest()->getParam('faq_id');
+
+        if ($id) {
+            $question = $this->faqRepository->getById((int)$id);
+            $question->delete();
+        }
```

Comment:
Koristiš `delete()` metodu koja je deprecated. Koristi repository, možeš u njega dodati metodu `deleteById()`.

ja bi ovako napravio + dodao bi još i try/catch:
```suggestion
        if ($id = $this->getRequest()->getParam('faq_id')) {
            $question = $this->faqRepository->getById((int)$id);
            $this->faqRepository->delete($question);
        }
```

Reply:
napravljene izmjene i stavljeno sve u try catch